src/validator.cpp

```diff
@@ -1207,12 +1221,9 @@ void Validator::ValidatorImpl::validateEquivalenceUnits(const ModelPtr &model, c
                                    });
             if (it == alreadyReported.end()) {
                 VariablePairPtr pair = VariablePair::create(variable, equivalentVariable);
-                ComponentPtr parentComponent = owningComponent(variable);
                 alreadyReported.push_back(pair);
-                auto unitsName = variable->units() == nullptr ? "" : variable->units()->name();
-                auto equivalentUnitsName = equivalentVariable->units() == nullptr ? "" : equivalentVariable->units()->name();
                 IssuePtr err = Issue::create();
-                err->setDescription("Variable '" + variable->name() + "' in component '" + parentComponent->name() + "' has units of '" + unitsName + "' and an equivalent variable '" + equivalentVariable->name() + "' in component '" + equivalentComponent->name() + "' with non-matching units of '" + equivalentUnitsName + "'. The mismatch is: " + hints);
+                err->setDescription("Variable '" + variable->name() + "' in component '" + parentComponent->name() + "' has units of '" + variable->units()->name() + "' and an equivalent variable '" + equivalentVariable->name() + "' in component '" + equivalentComponent->name() + "' with non-matching units of '" + equivalentVariable->units()->name() + "'. The mismatch is: " + hints);
```

Comment:
"... and its..." rather than "... and an..."?

Reply:
disagree. "...and an..." is correct in this context. There can be many equivalent variables...

Reply:
I agree there can be many equivalent variables, but I still thought it sounded better with "an" rather than with "its". However, you guys are the native English speakers, so you ought to know best. :)